crates/wit-component/tests/components/worlds-with-type-renamings/component.wat

```diff
@@ -57,7 +56,7 @@
   )
   (instance (;1;) (instantiate 0
       (with "import-func-the-func" (func 1))
-      (with "import-type-some-type" (type 3))
+      (with "import-type-some-type" (type 1))
```

Comment:
The `(;1;)` syntax is a way of saying that the item is introducing something new and 1 is the index of what was introduced. If it had a name (like `$foo`) that'd be printed instead. Here though no new index space items are being introduced, this is instead a reference to type 1 which is part of the instantiation (which creates instance 1 in this case)

Reply:
Oh oops sorry for misunderstanding!

As for whether or not this change works, it should be ok. Everything related to types pre-resources doesn't really matter a whole long "so long as it typechecks" so where exactly `(record)` comes from doesn't actually matter a whole lot.

As for whether this is expected, I'll say from my point of view the answer is "well as long as it works I guess". My overall strategy right now is to implement resources in the `roundtrip-wit` fuzzer and then just run it until everything works. That ideally stresses everything in `wit-component` and `wit-parser` to the extent where when it's done you should be able to take any WIT file and turn it into a real, valid, component. My strategy here has not really been all that principled of meticulously making sure components have an exact shape but moreso saying "well if the fuzzers and the tests pass it's probably ok" in the sense where further tweaks to the internals of a component can always change in the future as necessary (theoretically at least).

So I think the answer to your question is basically that while I did not intend for this change to the output component here to happen it's one that I generally ignore. All the various bits and bobs of tooling still ingest this as expected and process it successfully, so it's meeting my personal threshold at least for "probably good enough for now".

Now I do realize, though, that this isn't a great answer because it means a lot of stuff is probably slipping through the cracks. That's where my hope is that as a tooling-level concern `wit-component` is more flexible than a piece of the puzzle than `wasmparser` for example which has very little tolerance for mistakes.

Reply:
Ok, then I won't worry about it.